ReactiveTask/Task.swift

```diff
@@ -199,44 +202,47 @@ private final class Pipe {
 	/// anywhere else, as it may close unexpectedly.
 	///
 	/// Returns a producer that will complete or error.
-	func writeDataFromProducer(producer: SignalProducer<NSData, NoError>) -> SignalProducer<(), TaskError> {
+	func writeDataFromProducer(_ producer: SignalProducer<Data, NoError>) -> SignalProducer<(), TaskError> {
 		return SignalProducer { observer, disposable in
-			dispatch_group_enter(self.group)
-			let channel = dispatch_io_create(DISPATCH_IO_STREAM, self.writeFD, self.queue) { error in
+			self.group.enter()
+			let channel = DispatchIO(type: .stream, fileDescriptor: self.writeFD, queue: self.queue) { error in
 				if error == 0 {
 					observer.sendCompleted()
 				} else if error == ECANCELED {
 					observer.sendInterrupted()
 				} else {
-					observer.sendFailed(.POSIXError(error))
+					observer.sendFailed(.posixError(error))
 				}
 
 				close(self.writeFD)
-				dispatch_group_leave(self.group)
+				self.group.leave()
 			}
 
 			producer.startWithSignal { signal, producerDisposable in
-				disposable.addDisposable(producerDisposable)
+				disposable.add(producerDisposable)
 
 				signal.observe(Observer(next: { data in
-					let dispatchData = dispatch_data_create(data.bytes, data.length, self.queue, nil)
-
-					dispatch_io_write(channel, 0, dispatchData, self.queue) { (done, data, error) in
+					let dispatchData = data.withUnsafeBytes { (bytes: UnsafePointer<UInt8>) -> DispatchData in
+						let buffer = UnsafeBufferPointer(start: bytes, count: data.count)
+						return DispatchData(bytesNoCopy: buffer, deallocator: .custom(nil, {}))
```

Comment:
I'd much prefer to not use `bytesNoCopy`, but using `DispatchData(bytes: buffer)` crashes with EXC_BAD_ACCESS when the DispatchData deallocates.